tst/testinstall/list.tst

```diff
@@ -0,0 +1,201 @@
+gap> l := ListWithIdenticalEntries(0, 'w');
+""
+gap> IsStringRep(l);
+true
+gap> l := ListWithIdenticalEntries(0, true);
+[  ]
+gap> IsBlistRep(l);
+true
+gap> l := ListWithIdenticalEntries(0, fail);
+[  ]
+gap> IsPlistRep(l);
+true
+
+# ListWithIdenticalEntries: strings
+gap> l := ListWithIdenticalEntries(1, 'y');
+"y"
+gap> IsStringRep(l);
+true
+gap> l := ListWithIdenticalEntries(2, 'z');
+"zz"
+gap> IsStringRep(l);
+true
+gap> l := ListWithIdenticalEntries(76, '#');
+"############################################################################"
+gap> IsStringRep(l);
+true
+
+# ListWithIdenticalEntries: blists
+gap> l := ListWithIdenticalEntries(1, true);
+[ true ]
+gap> IsBlistRep(l);
+true
+gap> l := ListWithIdenticalEntries(2, true);
+[ true, true ]
+gap> IsBlistRep(l);
+true
+gap> l := ListWithIdenticalEntries(3, true);
+[ true, true, true ]
+gap> IsBlistRep(l);
+true
+gap> ForAll([1 .. 100],
+>           i -> ForAll(ListWithIdenticalEntries(i, true), x -> x));
+true
+gap> l := ListWithIdenticalEntries(1, false);
+[ false ]
+gap> IsBlistRep(l);
+true
+gap> l := ListWithIdenticalEntries(2, false);
+[ false, false ]
+gap> IsBlistRep(l);
+true
+gap> l := ListWithIdenticalEntries(3, false);
+[ false, false, false ]
+gap> IsBlistRep(l);
+true
+gap> ForAny([1 .. 100],
+>           i -> ForAny(ListWithIdenticalEntries(i, false), x -> x));
+false
+
+# ListWithIdenticalEntries: other
+gap> l := ListWithIdenticalEntries(2, Group(()));
+[ Group(()), Group(()) ]
+gap> IsPlistRep(l);
+true
+gap> IsIdenticalObj(l[1], l[2]);
+true
+gap> l := ListWithIdenticalEntries(10, "GAP");;
+gap> TNUM_OBJ(l);
+[ 36, "list (plain,hom)" ]
+gap> l;
+[ "GAP", "GAP", "GAP", "GAP", "GAP", "GAP", "GAP", "GAP", "GAP", "GAP" ]
+gap> TNUM_OBJ(l);
+[ 42, "list (plain,table)" ]
+gap> l := ListWithIdenticalEntries(10, PrimitiveRoot(GF(5)));
+[ Z(5), Z(5), Z(5), Z(5), Z(5), Z(5), Z(5), Z(5), Z(5), Z(5) ]
+gap> TNUM_OBJ(l);
+[ 60, "list (sml fin fld elms)" ]
+gap> l := ListWithIdenticalEntries(10, 5 / 7);
+[ 5/7, 5/7, 5/7, 5/7, 5/7, 5/7, 5/7, 5/7, 5/7, 5/7 ]
+gap> TNUM_OBJ(l);
+[ 54, "list (plain,cyc)" ]
+gap> l := ListWithIdenticalEntries(5, -1);
+[ -1, -1, -1, -1, -1 ]
+gap> TNUM_OBJ(l);
+[ 54, "list (plain,cyc)" ]
+gap> l := ListWithIdenticalEntries(5, 8);
+[ 8, 8, 8, 8, 8 ]
+gap> TNUM_OBJ(l);
+[ 54, "list (plain,cyc)" ]
+gap> l := ListWithIdenticalEntries(5, 0);
+[ 0, 0, 0, 0, 0 ]
+gap> TNUM_OBJ(l);
+[ 54, "list (plain,cyc)" ]
+gap> l := ListWithIdenticalEntries(5, infinity);
+[ infinity, infinity, infinity, infinity, infinity ]
+gap> TNUM_OBJ(l);
+[ 36, "list (plain,hom)" ]
+gap> l := ListWithIdenticalEntries(4, []);;
+gap> TNUM_OBJ(l);
+[ 36, "list (plain,hom)" ]
+gap> l;
+[ [  ], [  ], [  ], [  ] ]
+gap> TNUM_OBJ(l);
+[ 48, "list (plain,rect table)" ]
+gap> l := ListWithIdenticalEntries(4, [5]);;
+gap> TNUM_OBJ(l);
+[ 36, "list (plain,hom)" ]
+gap> l;
+[ [ 5 ], [ 5 ], [ 5 ], [ 5 ] ]
+gap> TNUM_OBJ(l);
+[ 48, "list (plain,rect table)" ]
```

Comment:
Those TNUMs change from time to time (in fact they've been changing around on master quite a bit this past year), making this test somewhat brittle. Perhaps use `TNUM_OBJ(l)[2];` instead of `TNUM_OBJ(l);`?

Anyway, we'll survive either way (so if this is left as-is, then the next time plist tnums change, we can still look into changing this then)